src/Stratis.Bitcoin/Connection/PeerBanning.cs

```diff
@@ -113,7 +113,7 @@ public void BanPeer(IPEndPoint endpoint, int banTimeSeconds, string reason = nul
             reason = reason ?? "unknown";
 
             bool banPeer = true;
-            INetworkPeer peer = this.connectionManager.ConnectedPeers.FindByEndpoint(endpoint);
+            INetworkPeer peer = this.networkPeerCollection.FindByEndpoint(endpoint);
```

Comment:
Who adds entries to this collection? The connection manager is the one responsible for having a list of connected peers so that other components can ask it and work with that list of connected peers. So the original implementation make a good sense, this does not. What am I missing?

Reply:
Yes we already discussed this, the PR will be changed to a different approach see the #1120 discussion

Reply:
I know, it is just that it seemed that this PR introduces a major bug, regardless of the approach, such bug must not be introduced

Reply:
Yes very true its from a lack of understanding of the code, we covered it when we did peer programming.

Reply:
Hi,  yes I'm going to revert the code and fix based on the discussion I had with Dan. Cheers.